Source/missiles.cpp

```diff
@@ -4550,6 +4651,7 @@ void missiles_process_charge()
 {
 	for (int i = 0; i < ActiveMissileCount; i++) {
 		int mi = ActiveMissiles[i];
+		auto &missile = Missiles[mi];
 		MissileStruct *mis = &Missiles[mi];

```

Comment:
Replace the uses of `mis->` with `missile.` and you can get rid of this pointer variable.